File: ts-sdk/src/sdk/models/operations/listbranches.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class ListBranchesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}


export class ListBranchesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}


export class ListBranchesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ListBranchesPathParams;

  @SpeakeasyMetadata()
  queryParams: ListBranchesQueryParams;
}


export class ListBranchesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  listBranches200ApplicationJSONObject?: Record<string, Record<string, any>>;
}

File: go-sdk/pkg/models/operations/listbranches.go
package operations

type ListBranchesPathParams struct {
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type ListBranchesQueryParams struct {
	Page    *float64 `queryParam:"style=form,explode=true,name=page"`
	PerPage *float64 `queryParam:"style=form,explode=true,name=per_page"`
}

type ListBranchesRequest struct {
	PathParams  ListBranchesPathParams
	QueryParams ListBranchesQueryParams
}

type ListBranchesResponse struct {
	ContentType                          string
	StatusCode                           int64
	ListBranches200ApplicationJSONObject map[string]map[string]interface{}
}
